### PR TITLE
Add basic documentation for jobs

### PR DESCRIPTION
**Pull request by zacwood9**
Alternative title: my Sunday afternoon figuring out how jobs work :-) 
I definitely plan on writing about this more in-depth in the future, but for the documentation I just included the basics that I got stuck on as I was trying to figure things out.

### CODE REVIEW

File: Guide/jobs.markdown [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+A pseudocode implementation of a job that sends a marketing email to all of the customers:
+
+```haskell
+module Web.Job.EmailCustomers where
+import Web.Controller.Prelude
+
+instance Job EmailCustomersJob where
+    perform EmailCustomersJob { .. } = do
+      customers <- query @Customer |> fetch
+      mapM_ (\customer -> sendMail (MarketingMail customer)) customers
```
Comment (mpscholten) [inline Guide/jobs.markdown:40]: I'd use `forEach` here which is like mapM_ but arguments reversed. This feels more like the IHP approach :)

**Review by mpscholten — COMMENTED** [review]
This is great, thanks :)

File: Guide/jobs.markdown (hunk 1 above)
Comment (zacwood9) [reply Guide/jobs.markdown]: Never thought of using `forEach` outside of views! Thanks

**Review by mpscholten — APPROVED** [review]
Perfect, thanks :)

**Pull request merged**